src/glados/utils/automated_install.py

```
@@ -0,0 +1,129 @@
+        subprocess.run(["ollama", "--version"], capture_output=True, text=True, check=True)
+        return True
+    except (subprocess.CalledProcessError, FileNotFoundError):
+        return False
+
+def install_ollama_windows():
+    """Download and install Ollama on Windows."""
+    download_url = "https://ollama.com/download/OllamaSetup.exe"
+    installer_path = Path("OllamaSetup.exe")
+    
+    try:
+        subprocess.run(["curl", "-L", download_url, "-o", str(installer_path)], check=True)
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)